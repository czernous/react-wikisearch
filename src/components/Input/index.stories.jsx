import React from 'react';

import Input from '.';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
    title: 'Components/Input',
    component: Input,
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <Input {...args} />;

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = {
    placeholder: 'Search data',
};
