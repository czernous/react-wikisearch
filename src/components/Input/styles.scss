.input-field {
    padding: 10px 20px;
    border: 1px solid gray;
    border-radius: 50px;
    font-size: 24px;
    width: 375px;

    &:focus {
        box-shadow: none;
        outline: none;
    }
}
