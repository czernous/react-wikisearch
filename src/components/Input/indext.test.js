import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Input from '.';

const renderer = new ShallowRenderer();

describe('Input component', () => {
    let sut;
    let props;

    describe('without props', () => {
        beforeEach(() => {
            sut = renderer.render(<Input className="input-field" {...props} />);
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        });
    });

    describe('with props', () => {
        beforeEach(() => {
            props = {
                placeholder: 'Enter your data',
                name: 'search bar',
            };

            sut = renderer.render(<Input className="input-field" {...props} />);
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        });
    });
});
