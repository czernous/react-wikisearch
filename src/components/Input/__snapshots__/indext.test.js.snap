// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component with props should match snapshot 1`] = `
<input
  className="input-field"
  name="search bar"
  placeholder="Enter your data"
/>
`;

exports[`Input component without props should match snapshot 1`] = `
<input
  className="input-field"
/>
`;
